swaywm diagonal window resizing broken, poor performance

I'm not sure I could describe the issue appropriately so I made a video: https://www.youtube.com/watch?v=jARlwPBSErg&feature=youtu.be

You can see the performance issue when resizing diagonally.  Horizontal/vertical resizing works just fine.

Using latest swaywm-git and wlroots-git

The type of windows I'm resizing doesn't seem to matter, but in this case it's xfce4-terminal and firefox-nightly (xwayland not being used).

> Atomic layout updates synchronize layout updates with all clients before rendering the next frame, so resizes look less choppy. (Well, normally...)
> 
> The issue here turned out to be easier to track down than expected, and I've opened #5744 to fix it. Could you give it a try?

I appreciate you tackling that so very quickly, but unfortunately I'm not knowledgeable enough to test it out.  I cloned the repo and applied the patch but I can't seem to build sway.

> wu% ninja -C build
ninja: Entering directory `build'
[2/4] Compiling C object sway/sway.p/desktop_xwayland.c.o
FAILED: sway/sway.p/desktop_xwayland.c.o 
cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Iprotocols -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/json-c -I/usr/include/libevdev-1.0/ -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/libdrm -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.5-8355884f (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/desktop_xwayland.c.o -MF sway/sway.p/desktop_xwayland.c.o.d -o sway/sway.p/desktop_xwayland.c.o -c ../sway/desktop/xwayland.c
../sway/desktop/xwayland.c: In function ‘unmanaged_handle_map’:
../sway/desktop/xwayland.c:77:33: error: ‘struct <anonymous>’ has no member named ‘set_geometry’
   77 |  wl_signal_add(&xsurface->events.set_geometry, &surface->set_geometry);
      |                                 ^
[3/4] Linking target swaybar/swaybar
ninja: build stopped: subcommand failed.

--- a/sway/desktop/transaction.c
+++ b/sway/desktop/transaction.c
@@ -358,11 +358,20 @@ static void transaction_progress_queue(void) {
 	// If there's a bunch of consecutive transactions which all apply to the
 	// same views, skip all except the last one.
 	while (server.transactions->length >= 2) {
-		struct sway_transaction *a = server.transactions->items[0];
-		struct sway_transaction *b = server.transactions->items[1];
-		if (transaction_same_nodes(a, b)) {
+		struct sway_transaction *txn = server.transactions->items[0];
+		struct sway_transaction *dup = NULL;
+
+		for (int i = 1; i < server.transactions->length; i++) {
+			struct sway_transaction *maybe_dup = server.transactions->items[i];
+			if (transaction_same_nodes(txn, maybe_dup)) {
+				dup = maybe_dup;
+				break;
+			}
+		}
+
+		if (dup) {
 			list_del(server.transactions, 0);
-			transaction_destroy(a);
+			transaction_destroy(txn);
 		} else {
 			break;
 		}

